pybossa/api/__init__.py

```diff
@@ -417,6 +420,82 @@ def task_progress(project_id=None, short_name=None):
     return Response(json.dumps(task_count_dict), mimetype="application/json")
 
 
+@jsonpify
+@login_required
+@blueprint.route('/preferences/<user_name>', methods=['GET'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def get_user_preferences(user_name):
```

Comment:
What is the reason of having two functions `get_user_preferences` and `update_user_preferences`, instead of having one function and use `methods` to distinguish GET/PUT? seems there are lots of duplicate code in two functions

Reply:
Thank you for the comment. This is for separation of concerns, two methods for two routes (each REST API route associates to a specific function), just to support readability.

The duplicated code could be moved to a helper function and shared between the two, but it seems minimal.